### ISSUE TITLE
Detect type instabilities

### ISSUE THREAD

**Issue by davidanthoff** [tonyhffong/Lint.jl#197]
Would be really neat if the linter could emit messages about type instable functions.

**Comment by TotalVerb** [comment]
Lint can do this to a limited extent. I anticipate it getting a bit better after https://github.com/tonyhffong/Lint.jl/pull/184 which will improve Lint's static type analysis.

**Comment by TotalVerb** [comment]
With Lint 0.4.0 (now tagged, awaiting merge into METADATA), an increased number of type instabilities are detected:

```julia
julia> lintstr("""
       function f()
           x = [1, 2]
           x = [1.0, 2.0]
           x
       end""")
1-element Array{Lint.LintMessage,1}:
 none:3 W545 x: previously used variable has apparent type Array{Int64,1}, but now assigned Array{Float64,1}
```

Obviously there is much more work to be done on this front, but the necessary infrastructure is in place.
